File: Web/models/post.js
var mongoose = require("mongoose");

var PostSchema = new mongoose.Schema({
    userId: String,
    username: String,
    displayName: String,
    body: String,
    date: {type: Date, default: Date.now},
    likes: [String],
    comments: [
        {
            username: String,
            userId: String,
            body: String,
            date: {type: Date, default: Date.now},
            likes: [String]
        }
    ],
    viewers: [String],
    views: {type: Number, default: 0},
});

module.exports = mongoose.model("post", PostSchema, "post");

File: Mobile/SleepProject/app/src/main/java/com/andrewkir/sleepproject/Post.kt
package com.andrewkir.sleepproject

class Post {
    var userId = ""
    var username = ""
    var displayName = ""
    var body = ""
    var postTitle = ""
    var date = ""
    var likes = ArrayList<String>()
    var comments = ArrayList<Comment>()
    var viewers = ArrayList<String>()
    var views = Int
}

class Comment(commenterId: String, commentBody: String, date: String){
    var commenterId = commenterId
    var commentBody = commentBody
    var date = date
}